### DISCUSSION TITLE
wails带来了新的可能性，我们正在尝试迁移！

### DISCUSSION THREAD

**Discussion by geekroscom** [wailsapp/wails#2296] [Ideas]
我们的产品经历了Electron、VSCODE Extension的多次重构，这几天才有发现Wails，非常兴奋，目前我们正在尝试产品的重构。这个过程中有一些想法，看看是否可以在未来有所更新和支持！

我记得Node是可以生成.node文件用于生产环境调用，向之前接触的serialport等好像就是这么处理的，如果Wails能在Go和Node中间建立一个合理的桥梁，那么Node生态是否就可以直接使用了？他将有利于应用访问和操作系统层。

**Comment by geekroscom** [comment]
Go也可以很好的操作系统层，比如本地文件系统、USB设备等资源还是Node比较丰富。

**Comment by alanchenchen** [comment]
node能做的go都行，操作系统特定的GUI API怕是只能通过c绑定，另，,node文件其实就是个dll之类的，你也可以把go包编译成c shared library，如果不考虑windows话，可以直接用go plugin。
> 建议把issue用英文翻一下，不然作者可能不太会回复。。

**Comment by crismcn** [comment]
多窗口目前还没有较好的方式实现，迁移需谨慎。托盘啥的  等待V3吧
